src/server/req_modify.c

```diff
@@ -895,10 +899,12 @@ req_modifyReservation(struct batch_request *preq)
 				break;
 		}
 
-		if (d == 0) 
+		if (d == 0)
 			presv->ri_wattr[RESV_ATR_duration].at_flags &= ~ATR_VFLAG_SET;
 
-		if (d == 1 && r == 1 && e == 1) {
+		if (d == 1 && e == 1 && r == 1) {
+			presv->ri_alter_state = presv->ri_wattr[RESV_ATR_state].at_val.at_long;
```

Comment:
Why are you setting the reservation state to being altered if you are rejecting the alter request?

Reply:
In resv_revert_alter_times() we evaluate what state the reservation must be set in after revert. Therefore we need to pass in RESV_BEING_ALTERED as state before revert so that the correct state will be evaluated and set after revert. If we do not pass in this the state is set to NO